src/handlers/shared/get-user-role.ts

```diff
@@ -2,19 +2,26 @@ import { Context } from "../../types";
 
 export async function getUserRole(context: Context, user: string): Promise<string> {
   const orgLogin = context.payload.organization?.login;
+  const { config } = context;
+  const { maxConcurrentTasks } = config.miscellaneous;
 
   if (!orgLogin) {
     throw new Error("Organization login not found in context payload.");
   }
 
-  try {
-    const response = await context.octokit.orgs.getMembershipForUser({
-      org: orgLogin,
-      username: user,
-    });
+  const response = await context.octokit.orgs.getMembershipForUser({
+    org: orgLogin,
+    username: user,
+  });
 
-    return response.data.role;
-  } catch (error) {
-    return "contributor";
+  const matchingUser = maxConcurrentTasks.find(({ role }) => role.toLowerCase() === response.data.role);
+
+  if (matchingUser) {
+    //chech if the current user role matches any of those defined in the config
+    return matchingUser.role;
```

Comment:
I think we can return `matchingUser` and `minLimitTask` which gives us the limit too